docs/USAGE-GUIDE.md

```diff
@@ -1339,6 +1339,37 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
 **s2n_config_add_ticket_crypto_key** adds session ticket key on the server side. It would be ideal to add new keys after every (encrypt_decrypt_key_lifetime_in_nanos/2) nanos because
 this will allow for gradual and linear transition of a key from encrypt-decrypt state to decrypt-only state.
 
+### s2n\_connection\_free\_handshake
+
+```c
+int s2n_connection_free_handshake(struct s2n_connection *conn);
+```
+
+**s2n_connection_free_handshake** wipes and releases buffers and memory
+allocated during the TLS handshake.  This function should be called after the
+handshake is successfully negotiated and logging or recording of handshake data
+is complete.
+
+### s2n\_connection\_release\_buffers
+
+```c
+int s2n_connection_release_buffers(struct s2n_connection *conn);
+```
+
+**s2n_connection_release_buffers** wipes and free the `in` and `out` buffers
+associated with a connection.  This function may be called when a connection is
+in keep-alive or idle state to reduce memory overhead of long lived connections.
+
+### s2n\_connection\_realloc\_buffers
+
+```c
+int s2n_connection_realloc_buffers(struct s2n_connection *conn);
```

Comment:
Should we do this automatically after `s2n_connection_release_buffers` on the following `s2n_recv` or `s2n_send` calls?

Reply:
that would be a nice little simplification.  by adding a size check to s2n_send or s2n_recv on the in and out buffer would allow us to move the reallocation to that code and eliminate this call.

Reply:
this actually makes things a little better, as we wouldn't need to preallocate in `s2n_connection_wipe`.  What is more, we can wipe these buffers on `s2n_connection_close` too, i believe.